### PR TITLE
quic: return a more meaningful error when dialing QUIC draft-29

### LINKED ISSUE #2515: Better error message when trying to dial `/quic` peer
Using `v0.30.0`:
Currently `.Error()` gives:
```
Error: failed to dial: failed to dial $PEERID:
  * [/ip4/$IP/udp/4001/quic] unknown QUIC version
```
I would rather see something along the lines of:
```
Error: failed to dial: failed to dial $PEERID:
  * [/ip4/$IP/udp/4001/quic] QUIC draft 29 has been removed, QUIC RFC9000 is accessible with /quic-v1
```
I think a self explanatory error message is warranted here because this used to work and now it don't.
Note that this will be shown if all dials fail.

### PR DESCRIPTION
**Pull request by marten-seemann**
Fixes #2515.

### CODE REVIEW

File: p2p/transport/quicreuse/quic_multiaddr.go [hunk 1]
```diff
@@ -21,6 +23,8 @@ func ToQuicMultiaddr(na net.Addr, version quic.VersionNumber) (ma.Multiaddr, err
 	switch version {
 	case quic.Version1:
 		return udpMA.Encapsulate(quicV1MA), nil
+	case quicDraft29:
```
Comment (marten-seemann) [inline p2p/transport/quicreuse/quic_multiaddr.go:26]: Not sure if this is needed here, but it certainly doesn't hurt.

**Review by Jorropo — APPROVED** [review]
SGTM thx

**Comment by marten-seemann** [comment]
I'm not sure if this fix will still work with #2520. But that's a bug that we need to fix.

**Review by MarcoPolo — APPROVED** [review]

**Comment by marten-seemann** [comment]
With #2520, this won't work any more, because it's not the QUIC transport's responsibility any more to handle /quic addresses. Instead, the swarm would now return an `ErrNoGoodAddresses`. I'm not sure if we should change that, since this an exported error and users might do an error assertion.

@Jorropo, any thoughts?

**Comment by marten-seemann** [comment]
Closing. We'll solve this in the swarm after #2437 has landed.

**Pull request closed without merging**